File: address-geocode-us-python-examples/get_distance_to_water_soap_sdk_example.py

import sys
import os

sys.path.insert(0, os.path.abspath("../address-geocode-us-python/SOAP"))

from get_distance_to_water_soap import GetDistanceToWaterSoap

def get_distance_to_water_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n--------------------------------------------------")
    print("Address Geocode US - GetDistanceToWater - SOAP SDK")
    print("--------------------------------------------------")

    latitude = "34.419120"
    longitude = "-119.703421"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Latitude        : {latitude}")
    print(f"Longitude       : {longitude}")
    print(f"License Key     : {license_key}")
    print(f"Is Live         : {is_live}")
    print(f"Timeout Seconds : {timeout_seconds}")

    try:
        service = GetDistanceToWaterSoap(latitude,longitude,license_key,is_live,timeout_seconds)
        response = service.get_distance_to_water()

        if response and not hasattr(response, "Error"):
            print("\r\n* Distance Info *\r\n")
            print(f"Miles To Water          : {response.DistanceToWater}")
            print(f"Latitude                : {response.Latitude}")
            print(f"Longitude               : {response.Longitude}")
            print(f"Closest Water Latitude  : {response.WaterLat}")
            print(f"Closest Water Longitude : {response.WaterLon}")
        elif hasattr(response, "Error") and response.Error:
            print("\n* Error *\n")
            print(f"Error Description: {response.Error.Desc}")
            print(f"Error Number     : {response.Error.Number}")
            print(f"Error Location   : {response.Error.Location}")
        else:
            print("\nNo distance info found.")
    except Exception as e:
        print("\n* Exception *\n")
        print(f"Exception Message: {str(e)}")

File: address-geocode-us-dot-net-examples/GetDistanceToWaterSoapSdkExample.cs
using address_geocode_us_dot_net.SOAP;
using AGUSService;
using System.ComponentModel;

namespace address_geocode_us_dot_net_examples
{
    public static class GetDistanceToWaterSoapSdkExample
    {
        public static void Go(string LicenseKey, bool IsLive)
        {
            Console.WriteLine("\r\n--------------------------------------------------");
            Console.WriteLine("Address Geocode US - GetDistanceToWater - SOAP SDK");
            Console.WriteLine("--------------------------------------------------");

            string Latitude = "34.419120";
            string Longitude = "-119.703421";

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Latitude    : {Latitude}");
            Console.WriteLine($"Longitude   : {Longitude}");
            Console.WriteLine($"License Key : {LicenseKey}");
            Console.WriteLine($"Is Live     : {IsLive}");

            GetDistanceToWaterValidation getDistanceToWaterValidation = new(IsLive);
            DistanceToWaterInfo response = getDistanceToWaterValidation.GetDistanceToWater(Latitude, Longitude, LicenseKey).Result;

            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Distance To Water Info *\r\n");
                Console.WriteLine($"Distance To Water: {response.DistanceToWater}");
                Console.WriteLine($"Latitude         : {response.Latitude}");
                Console.WriteLine($"Longitude        : {response.Longitude}");
                Console.WriteLine($"Water Latitude   : {response.WaterLat}");
                Console.WriteLine($"Water Longitude  : {response.WaterLon}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Description: {response.Error.Desc}");
                Console.WriteLine($"Error Number     : {response.Error.Number}");
                Console.WriteLine($"Error Location   : {response.Error.Location}");
            }
        }
    }
}